gen2-human-machine-safety: improvements

ES-Alexander opened the pull request:
- improved clarity by reducing copied code and improving variable names
- vectorised `calc_spatials` depth averaging
- added easier support for alternative averaging methods (e.g. `np.median`)

VanDavv commented on gen2-human-machine-safety/main.py:
```diff
@@ -39,31 +44,26 @@ def __init__(self):
+        ymax -= deltaY
+
+        # Calculate the average depth in the ROI.
+        depthROI = depth[ymin:ymax, xmin:xmax]
```
Sometimes the detection bbox is flipped (e.g. `bbox` variable can be `[720, 720, 0, 0]`), so cutting the frame before check throws an error at random during runtime

VanDavv reviewed (changes requested):
@ES-Alexander Thanks for your PR! I found some issues, but your implementation on depth ROI calculation is definitely better. I went ahead quickly and self-addressed the comments below in this PR - https://github.com/luxonis/depthai-experiments/pull/212 - where you can find both your commit and my improvements

Luxonis-Brandon commented:
Thanks @ES-Alexander and @VanDavv !

ES-Alexander commented:
> I found some issues

Fair enough - I probably should have been more rigorous with, you know, actually testing it :-P
The depth averaging was linked to from somewhere (can't remember if it was discord or in the OAK-D-Lite kickstarter FAQs or something). I swung by to take a look and saw that it could be done better, and once I'd fixed that I saw a couple of other repetitive things that I thought I may as well fix while I was in there :-)

> I went ahead quickly and self-addressed the comments below in this PR - #212 - where you can find both your commit and my improvements

Thanks!
